ci/env/install-dependencies.sh

```diff
@@ -66,138 +66,7 @@ install_base() {
-        "${WORKSPACE_DIR}"/ci/suppress_output "${miniconda_target}" -f -b -p "${miniconda_dir}"
-        conda="${miniconda_dir}/bin/conda"
-        ;;
-    esac
-  fi
-
-  if [ ! -x "${CONDA_PYTHON_EXE-}" ]; then  # If conda isn't activated, activate it
-    local restore_shell_state=""
-    if [ -o xtrace ]; then set +x && restore_shell_state="set -x"; fi  # Disable set -x (noisy here)
-
-    # TODO(mehrdadn): conda activation is buggy on MSYS2; it adds C:/... to PATH,
-    # which gets split on a colon. Is it necessary to work around this?
-    eval "$("${conda}" shell."${SHELL##*/}" hook)"  # Activate conda
-    conda init "${SHELL##*/}"  # Add to future shells
-
-    ${restore_shell_state}  # Restore set -x
-  fi
-
-  local python_version
-  python_version="$(python -s -c "import sys; print('%s.%s' % sys.version_info[:2])")"
-  if [ -n "${PYTHON-}" ] && [ "${PYTHON}" != "${python_version}" ]; then  # Update Python version
-    (
-      set +x
-      echo "Updating Anaconda Python ${python_version} to ${PYTHON}..."
-      "${WORKSPACE_DIR}"/ci/suppress_output conda install -q -y python="${PYTHON}"
-    )
-  elif [ "${MINIMAL_INSTALL-}" = "1" ]; then  # Reset environment
-    (
-      set +x
-      echo "Resetting Anaconda Python ${python_version}..."
-      "${WORKSPACE_DIR}"/ci/suppress_output conda install -q -y --rev 0
-    )
-  fi
-
-  if [[ "${PYTHON-}" != "3.12" ]]; then
-    # Install mpi4py as a test dependency for Python <3.12; currently mpi4py is not 
-    # available for Python 3.12
-    "${WORKSPACE_DIR}"/ci/suppress_output conda install -c anaconda mpi4py -y
-  fi
-
-  command -V python
-  test -x "${CONDA_PYTHON_EXE}"  # make sure conda is activated
-}
-
-install_shellcheck() {
-  local shellcheck_version="0.7.1"
-  if [ "${shellcheck_version}" != "$(command -v shellcheck > /dev/null && shellcheck --version | sed -n "s/version: //p")" ]; then
-    local osname=""
-    case "${OSTYPE}" in
-      linux*) osname="linux";;
-      darwin*) osname="darwin";;
-    esac
-    local name="shellcheck-v${shellcheck_version}"
-    if [ "${osname}" = linux ] || [ "${osname}" = darwin ]; then
-      sudo mkdir -p /usr/local/bin || true
-      curl -f -s -L "https://github.com/koalaman/shellcheck/releases/download/v${shellcheck_version}/${name}.${osname}.x86_64.tar.xz" | {
-        sudo tar -C /usr/local/bin -x -v -J --strip-components=1 "${name}/shellcheck"
-      }
-    else
-      mkdir -p /usr/local/bin
-      curl -f -s -L -o "${name}.zip" "https://github.com/koalaman/shellcheck/releases/download/v${shellcheck_version}/${name}.zip"
-      unzip "${name}.zip" "${name}.exe"
-      mv -f "${name}.exe" "/usr/local/bin/shellcheck.exe"
-    fi
-    test "${shellcheck_version}" = "$(shellcheck --version | sed -n "s/version: //p")"
-  fi
-}
-
-install_linters() {
-  pip install -r "${WORKSPACE_DIR}"/python/requirements/lint-requirements.txt
-
-  install_shellcheck
+  source "${SCRIPT_DIR}/install-miniconda.sh"
```

Comment:
why not just bash run this script? why source is required? sourcing in a function is a bit weird..

if `bash install-miniconda.sh` does not work (it should work, otherwise, how does the `RUN` command work?), then you can move the function there, like:


````
#!/bin/bash

function install_miniconda {
   # function body
}

if [[ "${BASH_SOURCE[0]}" == "${0}" ]]; then install_miniconda ; fi

````